### PR TITLE
[rc-swift] Start RC Swift implementation

### PR DESCRIPTION
**Pull request by paulb777**
- CocoaPods setup sufficient to run mixed language unit tests
- RemoteConfigUpdate
- UserDefaultsManager

Note: The first pass of the implementation is not worrying about access levels. Extra public's and `@objc` declarations so that the ObjC unit tests can continue to run.

### CODE REVIEW

File: .github/workflows/remoteconfig.yml [hunk 1]
```diff
@@ -59,19 +59,15 @@ jobs:
...
         podspec: [FirebaseRemoteConfig.podspec]
         os: [macos-14]
         include:
-          - os: macos-14
-            xcode: Xcode_15.3
-            # TODO(#13078): Fix testing infra to enforce warnings again.
-            tests: --allow-warnings
           # Flaky tests on CI
           - os: macos-14
             xcode: Xcode_16
```
Comment (andrewheard) [inline .github/workflows/remoteconfig.yml:69]: We might want to get ahead of https://github.com/actions/runner-images/issues/10703 and switch this to `macos-15` here and in `main`.

**Review by andrewheard — APPROVED** [review]
> CocoaPods setup sufficient to run mixed language unit tests

Nice approach!

File: .github/workflows/remoteconfig.yml (hunk 1 above)
Comment (paulb777) [reply .github/workflows/remoteconfig.yml:69]: Agreed, but punting for now.

**Comment by paulb777** [comment]
Going to merge and can address any future comments in future `rc-swift` PRs

**Pull request merged**